examples/http_rama_tower.rs

```diff
@@ -0,0 +1,242 @@
+//! This example demonstrates how to integrate tower into your rama HTTP stack
+//!
+//! ```sh
+//! cargo run --example http_rama_tower --features=http-full,tower
+//! ```
+//!
+//! # Expected output
+//!
+//! The server will start and listen on `:62020`. You can use your browser to interact with the service:
+//!
+//! ```sh
+//! open http://127.0.0.1:62020
+//! curl -v http://127.0.0.1:62020
+//! ```
+//!
+//! You should see the homepage in your browser with the title "Rama + Tower".
+
+/// rama provides everything out of the box to build a complete web service.
+use rama::{
+    Layer as _,
+    error::{BoxError, OpaqueError},
+    http::{
+        HeaderValue, Request, layer::trace::TraceLayer, response::Html, server::HttpServer,
+        service::web::Router,
+    },
+    layer::ConsumeErrLayer,
+    net::address::SocketAddress,
+    rt::Executor,
+    utils::tower::{
+        ServiceAdapter,
+        core::{Layer, Service},
+        layer::LayerAdapter,
+    },
+};
+
+/// Everything else we need is provided by the standard library, community crates or tokio.
+use pin_project_lite::pin_project;
+use std::{
+    convert::Infallible,
+    future::{Ready, ready},
+    pin::Pin,
+    time::Duration,
+};
+use tokio::time::Sleep;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::{EnvFilter, fmt};
+
+const ADDRESS: SocketAddress = SocketAddress::local_ipv4(62020);
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::DEBUG.into())
+                .from_env_lossy(),
+        )
+        .init();
+
+    let graceful = rama::graceful::Shutdown::default();
+
+    let router: Router<()> = Router::new().get("/", ServiceAdapter::new(HelloSvc));
+    let app = LayerAdapter::new((Timeout(Duration::from_secs(30)), AddHelloMarkerHeader))
+        .into_layer(router);
+
+    graceful.spawn_task_fn(async |guard| {
+        tracing::info!("running service at: {ADDRESS}");
+        let exec = Executor::graceful(guard);
+        HttpServer::auto(exec)
+            .listen(
+                ADDRESS,
+                (TraceLayer::new_for_http(), ConsumeErrLayer::default()).into_layer(app),
+            )
+            .await
+            .unwrap();
+    });
+
+    graceful
+        .shutdown_with_limit(Duration::from_secs(30))
+        .await
+        .expect("graceful shutdown");
+}
+
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+struct AddHelloMarkerHeader;
```

Comment:
```suggestion
struct AddHelloMarkerHeaderLayer;
```
I would use layer inside of name, pretty confusing without it since it's the standard practice everywhere else in rama